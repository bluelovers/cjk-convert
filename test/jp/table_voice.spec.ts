import {
	getVoiceAll,
	EnumTableVoice,
	toVoice01,
	toVoice02,
	toVoice03,
	table_voice,
	ITableVoiceKeys,
} from '../../lib/jp/table_voice';
import { array_unique } from 'array-hyper-unique';

describe(`table_voice_keys`, () =>
{
	let table_voice_keys = Object.keys(table_voice) as ITableVoiceKeys[];

	table_voice_keys
		.forEach((char) => {

			const arr = getVoiceAll(char);
			const len = arr.length;

			// @ts-ignore
			it(`${char}.length 1 ~ 3`, function ()
			{
				expect(Array.isArray(arr)).toBeTruthy();
				expect(arr.length).toBeGreaterThan(1);
				expect(arr.length).toBeLessThanOrEqual(3);

				let actual = arr[EnumTableVoice['01']];
				let expected = toVoice01(char);

				expect(actual).toStrictEqual(expected)

				expect(array_unique(arr)).toHaveLength(len)

				expect(actual).toMatchSnapshot();
				expect(arr).toMatchSnapshot();
			});

			it(`${char} 01`, function ()
			{
				expect(toVoice01(char)).toStrictEqual(getVoiceAll(char)[EnumTableVoice['01']]);
				expect(EnumTableVoice['01']).toStrictEqual(0)
			});

			it(`${char} 02`, function ()
			{
				expect(toVoice02(char)).toStrictEqual(getVoiceAll(char)[EnumTableVoice['02']]);
				expect(EnumTableVoice['02']).toStrictEqual(1)
			});

			it(`${char} 03`, function ()
			{
				expect(toVoice03(char)).toStrictEqual(getVoiceAll(char)[EnumTableVoice['03']]);
				expect(EnumTableVoice['03']).toStrictEqual(2)
			});

		})
	;

})

describe(`getVoice`, () =>
{

	test(`か`, () =>
	{
		let char = 'か';

		let actual = getVoiceAll(char);

		expect(toVoice01(char)).toStrictEqual(char);
		expect(toVoice02(char)).toStrictEqual('が');
		expect(toVoice03(char)).toBeUndefined();

		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

})

describe(`throw error`, () =>
{

	test(`null`, () =>
	{
		expect(() => toVoice01(null)).toThrowError();
	});

	test(`undefined`, () =>
	{
		expect(() => toVoice01(void 0)).toThrowError();
	});

	test(`empty text`, () =>
	{
		expect(() => toVoice01('')).toThrowError();
	});

	test(`k`, () =>
	{
		expect(() => toVoice01('k')).not.toThrowError();
	});

})
