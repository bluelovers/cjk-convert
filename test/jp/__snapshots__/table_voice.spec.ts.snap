// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVoice か 1`] = `
Array [
  "か",
  "が",
]
`;

exports[`table_voice_keys か.length 1 ~ 3 1`] = `"か"`;

exports[`table_voice_keys か.length 1 ~ 3 2`] = `
Array [
  "か",
  "が",
]
`;

exports[`table_voice_keys が.length 1 ~ 3 1`] = `"か"`;

exports[`table_voice_keys が.length 1 ~ 3 2`] = `
Array [
  "か",
  "が",
]
`;

exports[`table_voice_keys き.length 1 ~ 3 1`] = `"き"`;

exports[`table_voice_keys き.length 1 ~ 3 2`] = `
Array [
  "き",
  "ぎ",
]
`;

exports[`table_voice_keys ぎ.length 1 ~ 3 1`] = `"き"`;

exports[`table_voice_keys ぎ.length 1 ~ 3 2`] = `
Array [
  "き",
  "ぎ",
]
`;

exports[`table_voice_keys く.length 1 ~ 3 1`] = `"く"`;

exports[`table_voice_keys く.length 1 ~ 3 2`] = `
Array [
  "く",
  "ぐ",
]
`;

exports[`table_voice_keys ぐ.length 1 ~ 3 1`] = `"く"`;

exports[`table_voice_keys ぐ.length 1 ~ 3 2`] = `
Array [
  "く",
  "ぐ",
]
`;

exports[`table_voice_keys け.length 1 ~ 3 1`] = `"け"`;

exports[`table_voice_keys け.length 1 ~ 3 2`] = `
Array [
  "け",
  "げ",
]
`;

exports[`table_voice_keys げ.length 1 ~ 3 1`] = `"け"`;

exports[`table_voice_keys げ.length 1 ~ 3 2`] = `
Array [
  "け",
  "げ",
]
`;

exports[`table_voice_keys こ.length 1 ~ 3 1`] = `"こ"`;

exports[`table_voice_keys こ.length 1 ~ 3 2`] = `
Array [
  "こ",
  "ご",
]
`;

exports[`table_voice_keys ご.length 1 ~ 3 1`] = `"こ"`;

exports[`table_voice_keys ご.length 1 ~ 3 2`] = `
Array [
  "こ",
  "ご",
]
`;

exports[`table_voice_keys さ.length 1 ~ 3 1`] = `"さ"`;

exports[`table_voice_keys さ.length 1 ~ 3 2`] = `
Array [
  "さ",
  "ざ",
]
`;

exports[`table_voice_keys ざ.length 1 ~ 3 1`] = `"さ"`;

exports[`table_voice_keys ざ.length 1 ~ 3 2`] = `
Array [
  "さ",
  "ざ",
]
`;

exports[`table_voice_keys し.length 1 ~ 3 1`] = `"し"`;

exports[`table_voice_keys し.length 1 ~ 3 2`] = `
Array [
  "し",
  "じ",
]
`;

exports[`table_voice_keys じ.length 1 ~ 3 1`] = `"し"`;

exports[`table_voice_keys じ.length 1 ~ 3 2`] = `
Array [
  "し",
  "じ",
]
`;

exports[`table_voice_keys す.length 1 ~ 3 1`] = `"す"`;

exports[`table_voice_keys す.length 1 ~ 3 2`] = `
Array [
  "す",
  "ず",
]
`;

exports[`table_voice_keys ず.length 1 ~ 3 1`] = `"す"`;

exports[`table_voice_keys ず.length 1 ~ 3 2`] = `
Array [
  "す",
  "ず",
]
`;

exports[`table_voice_keys せ.length 1 ~ 3 1`] = `"せ"`;

exports[`table_voice_keys せ.length 1 ~ 3 2`] = `
Array [
  "せ",
  "ぜ",
]
`;

exports[`table_voice_keys ぜ.length 1 ~ 3 1`] = `"せ"`;

exports[`table_voice_keys ぜ.length 1 ~ 3 2`] = `
Array [
  "せ",
  "ぜ",
]
`;

exports[`table_voice_keys そ.length 1 ~ 3 1`] = `"そ"`;

exports[`table_voice_keys そ.length 1 ~ 3 2`] = `
Array [
  "そ",
  "ぞ",
]
`;

exports[`table_voice_keys ぞ.length 1 ~ 3 1`] = `"そ"`;

exports[`table_voice_keys ぞ.length 1 ~ 3 2`] = `
Array [
  "そ",
  "ぞ",
]
`;

exports[`table_voice_keys た.length 1 ~ 3 1`] = `"た"`;

exports[`table_voice_keys た.length 1 ~ 3 2`] = `
Array [
  "た",
  "だ",
]
`;

exports[`table_voice_keys だ.length 1 ~ 3 1`] = `"た"`;

exports[`table_voice_keys だ.length 1 ~ 3 2`] = `
Array [
  "た",
  "だ",
]
`;

exports[`table_voice_keys ち.length 1 ~ 3 1`] = `"ち"`;

exports[`table_voice_keys ち.length 1 ~ 3 2`] = `
Array [
  "ち",
  "ぢ",
]
`;

exports[`table_voice_keys ぢ.length 1 ~ 3 1`] = `"ち"`;

exports[`table_voice_keys ぢ.length 1 ~ 3 2`] = `
Array [
  "ち",
  "ぢ",
]
`;

exports[`table_voice_keys つ.length 1 ~ 3 1`] = `"つ"`;

exports[`table_voice_keys つ.length 1 ~ 3 2`] = `
Array [
  "つ",
  "づ",
]
`;

exports[`table_voice_keys づ.length 1 ~ 3 1`] = `"つ"`;

exports[`table_voice_keys づ.length 1 ~ 3 2`] = `
Array [
  "つ",
  "づ",
]
`;

exports[`table_voice_keys て.length 1 ~ 3 1`] = `"て"`;

exports[`table_voice_keys て.length 1 ~ 3 2`] = `
Array [
  "て",
  "で",
]
`;

exports[`table_voice_keys で.length 1 ~ 3 1`] = `"て"`;

exports[`table_voice_keys で.length 1 ~ 3 2`] = `
Array [
  "て",
  "で",
]
`;

exports[`table_voice_keys と.length 1 ~ 3 1`] = `"と"`;

exports[`table_voice_keys と.length 1 ~ 3 2`] = `
Array [
  "と",
  "ど",
]
`;

exports[`table_voice_keys ど.length 1 ~ 3 1`] = `"と"`;

exports[`table_voice_keys ど.length 1 ~ 3 2`] = `
Array [
  "と",
  "ど",
]
`;

exports[`table_voice_keys は.length 1 ~ 3 1`] = `"は"`;

exports[`table_voice_keys は.length 1 ~ 3 2`] = `
Array [
  "は",
  "ば",
  "ぱ",
]
`;

exports[`table_voice_keys ば.length 1 ~ 3 1`] = `"は"`;

exports[`table_voice_keys ば.length 1 ~ 3 2`] = `
Array [
  "は",
  "ば",
  "ぱ",
]
`;

exports[`table_voice_keys ぱ.length 1 ~ 3 1`] = `"は"`;

exports[`table_voice_keys ぱ.length 1 ~ 3 2`] = `
Array [
  "は",
  "ば",
  "ぱ",
]
`;

exports[`table_voice_keys ひ.length 1 ~ 3 1`] = `"ひ"`;

exports[`table_voice_keys ひ.length 1 ~ 3 2`] = `
Array [
  "ひ",
  "び",
  "ぴ",
]
`;

exports[`table_voice_keys び.length 1 ~ 3 1`] = `"ひ"`;

exports[`table_voice_keys び.length 1 ~ 3 2`] = `
Array [
  "ひ",
  "び",
  "ぴ",
]
`;

exports[`table_voice_keys ぴ.length 1 ~ 3 1`] = `"ひ"`;

exports[`table_voice_keys ぴ.length 1 ~ 3 2`] = `
Array [
  "ひ",
  "び",
  "ぴ",
]
`;

exports[`table_voice_keys ふ.length 1 ~ 3 1`] = `"ふ"`;

exports[`table_voice_keys ふ.length 1 ~ 3 2`] = `
Array [
  "ふ",
  "ぶ",
  "ぷ",
]
`;

exports[`table_voice_keys ぶ.length 1 ~ 3 1`] = `"ふ"`;

exports[`table_voice_keys ぶ.length 1 ~ 3 2`] = `
Array [
  "ふ",
  "ぶ",
  "ぷ",
]
`;

exports[`table_voice_keys ぷ.length 1 ~ 3 1`] = `"ふ"`;

exports[`table_voice_keys ぷ.length 1 ~ 3 2`] = `
Array [
  "ふ",
  "ぶ",
  "ぷ",
]
`;

exports[`table_voice_keys へ.length 1 ~ 3 1`] = `"へ"`;

exports[`table_voice_keys へ.length 1 ~ 3 2`] = `
Array [
  "へ",
  "べ",
  "ぺ",
]
`;

exports[`table_voice_keys べ.length 1 ~ 3 1`] = `"へ"`;

exports[`table_voice_keys べ.length 1 ~ 3 2`] = `
Array [
  "へ",
  "べ",
  "ぺ",
]
`;

exports[`table_voice_keys ぺ.length 1 ~ 3 1`] = `"へ"`;

exports[`table_voice_keys ぺ.length 1 ~ 3 2`] = `
Array [
  "へ",
  "べ",
  "ぺ",
]
`;

exports[`table_voice_keys ほ.length 1 ~ 3 1`] = `"ほ"`;

exports[`table_voice_keys ほ.length 1 ~ 3 2`] = `
Array [
  "ほ",
  "ぼ",
  "ぽ",
]
`;

exports[`table_voice_keys ぼ.length 1 ~ 3 1`] = `"ほ"`;

exports[`table_voice_keys ぼ.length 1 ~ 3 2`] = `
Array [
  "ほ",
  "ぼ",
  "ぽ",
]
`;

exports[`table_voice_keys ぽ.length 1 ~ 3 1`] = `"ほ"`;

exports[`table_voice_keys ぽ.length 1 ~ 3 2`] = `
Array [
  "ほ",
  "ぼ",
  "ぽ",
]
`;

exports[`table_voice_keys カ.length 1 ~ 3 1`] = `"カ"`;

exports[`table_voice_keys カ.length 1 ~ 3 2`] = `
Array [
  "カ",
  "ガ",
]
`;

exports[`table_voice_keys ガ.length 1 ~ 3 1`] = `"カ"`;

exports[`table_voice_keys ガ.length 1 ~ 3 2`] = `
Array [
  "カ",
  "ガ",
]
`;

exports[`table_voice_keys キ.length 1 ~ 3 1`] = `"キ"`;

exports[`table_voice_keys キ.length 1 ~ 3 2`] = `
Array [
  "キ",
  "ギ",
]
`;

exports[`table_voice_keys ギ.length 1 ~ 3 1`] = `"キ"`;

exports[`table_voice_keys ギ.length 1 ~ 3 2`] = `
Array [
  "キ",
  "ギ",
]
`;

exports[`table_voice_keys ク.length 1 ~ 3 1`] = `"ク"`;

exports[`table_voice_keys ク.length 1 ~ 3 2`] = `
Array [
  "ク",
  "グ",
]
`;

exports[`table_voice_keys グ.length 1 ~ 3 1`] = `"ク"`;

exports[`table_voice_keys グ.length 1 ~ 3 2`] = `
Array [
  "ク",
  "グ",
]
`;

exports[`table_voice_keys ケ.length 1 ~ 3 1`] = `"ケ"`;

exports[`table_voice_keys ケ.length 1 ~ 3 2`] = `
Array [
  "ケ",
  "ゲ",
]
`;

exports[`table_voice_keys ゲ.length 1 ~ 3 1`] = `"ケ"`;

exports[`table_voice_keys ゲ.length 1 ~ 3 2`] = `
Array [
  "ケ",
  "ゲ",
]
`;

exports[`table_voice_keys コ.length 1 ~ 3 1`] = `"コ"`;

exports[`table_voice_keys コ.length 1 ~ 3 2`] = `
Array [
  "コ",
  "ゴ",
]
`;

exports[`table_voice_keys ゴ.length 1 ~ 3 1`] = `"コ"`;

exports[`table_voice_keys ゴ.length 1 ~ 3 2`] = `
Array [
  "コ",
  "ゴ",
]
`;

exports[`table_voice_keys サ.length 1 ~ 3 1`] = `"サ"`;

exports[`table_voice_keys サ.length 1 ~ 3 2`] = `
Array [
  "サ",
  "ザ",
]
`;

exports[`table_voice_keys ザ.length 1 ~ 3 1`] = `"サ"`;

exports[`table_voice_keys ザ.length 1 ~ 3 2`] = `
Array [
  "サ",
  "ザ",
]
`;

exports[`table_voice_keys シ.length 1 ~ 3 1`] = `"シ"`;

exports[`table_voice_keys シ.length 1 ~ 3 2`] = `
Array [
  "シ",
  "ジ",
]
`;

exports[`table_voice_keys ジ.length 1 ~ 3 1`] = `"シ"`;

exports[`table_voice_keys ジ.length 1 ~ 3 2`] = `
Array [
  "シ",
  "ジ",
]
`;

exports[`table_voice_keys ス.length 1 ~ 3 1`] = `"ス"`;

exports[`table_voice_keys ス.length 1 ~ 3 2`] = `
Array [
  "ス",
  "ズ",
]
`;

exports[`table_voice_keys ズ.length 1 ~ 3 1`] = `"ス"`;

exports[`table_voice_keys ズ.length 1 ~ 3 2`] = `
Array [
  "ス",
  "ズ",
]
`;

exports[`table_voice_keys セ.length 1 ~ 3 1`] = `"セ"`;

exports[`table_voice_keys セ.length 1 ~ 3 2`] = `
Array [
  "セ",
  "ゼ",
]
`;

exports[`table_voice_keys ゼ.length 1 ~ 3 1`] = `"セ"`;

exports[`table_voice_keys ゼ.length 1 ~ 3 2`] = `
Array [
  "セ",
  "ゼ",
]
`;

exports[`table_voice_keys ソ.length 1 ~ 3 1`] = `"ソ"`;

exports[`table_voice_keys ソ.length 1 ~ 3 2`] = `
Array [
  "ソ",
  "ゾ",
]
`;

exports[`table_voice_keys ゾ.length 1 ~ 3 1`] = `"ソ"`;

exports[`table_voice_keys ゾ.length 1 ~ 3 2`] = `
Array [
  "ソ",
  "ゾ",
]
`;

exports[`table_voice_keys タ.length 1 ~ 3 1`] = `"タ"`;

exports[`table_voice_keys タ.length 1 ~ 3 2`] = `
Array [
  "タ",
  "ダ",
]
`;

exports[`table_voice_keys ダ.length 1 ~ 3 1`] = `"タ"`;

exports[`table_voice_keys ダ.length 1 ~ 3 2`] = `
Array [
  "タ",
  "ダ",
]
`;

exports[`table_voice_keys チ.length 1 ~ 3 1`] = `"チ"`;

exports[`table_voice_keys チ.length 1 ~ 3 2`] = `
Array [
  "チ",
  "ヂ",
]
`;

exports[`table_voice_keys ヂ.length 1 ~ 3 1`] = `"チ"`;

exports[`table_voice_keys ヂ.length 1 ~ 3 2`] = `
Array [
  "チ",
  "ヂ",
]
`;

exports[`table_voice_keys ツ.length 1 ~ 3 1`] = `"ツ"`;

exports[`table_voice_keys ツ.length 1 ~ 3 2`] = `
Array [
  "ツ",
  "ヅ",
]
`;

exports[`table_voice_keys ヅ.length 1 ~ 3 1`] = `"ツ"`;

exports[`table_voice_keys ヅ.length 1 ~ 3 2`] = `
Array [
  "ツ",
  "ヅ",
]
`;

exports[`table_voice_keys テ.length 1 ~ 3 1`] = `"テ"`;

exports[`table_voice_keys テ.length 1 ~ 3 2`] = `
Array [
  "テ",
  "デ",
]
`;

exports[`table_voice_keys デ.length 1 ~ 3 1`] = `"テ"`;

exports[`table_voice_keys デ.length 1 ~ 3 2`] = `
Array [
  "テ",
  "デ",
]
`;

exports[`table_voice_keys ト.length 1 ~ 3 1`] = `"ト"`;

exports[`table_voice_keys ト.length 1 ~ 3 2`] = `
Array [
  "ト",
  "ド",
]
`;

exports[`table_voice_keys ド.length 1 ~ 3 1`] = `"ト"`;

exports[`table_voice_keys ド.length 1 ~ 3 2`] = `
Array [
  "ト",
  "ド",
]
`;

exports[`table_voice_keys ハ.length 1 ~ 3 1`] = `"ハ"`;

exports[`table_voice_keys ハ.length 1 ~ 3 2`] = `
Array [
  "ハ",
  "バ",
  "パ",
]
`;

exports[`table_voice_keys バ.length 1 ~ 3 1`] = `"ハ"`;

exports[`table_voice_keys バ.length 1 ~ 3 2`] = `
Array [
  "ハ",
  "バ",
  "パ",
]
`;

exports[`table_voice_keys パ.length 1 ~ 3 1`] = `"ハ"`;

exports[`table_voice_keys パ.length 1 ~ 3 2`] = `
Array [
  "ハ",
  "バ",
  "パ",
]
`;

exports[`table_voice_keys ヒ.length 1 ~ 3 1`] = `"ヒ"`;

exports[`table_voice_keys ヒ.length 1 ~ 3 2`] = `
Array [
  "ヒ",
  "ビ",
  "ピ",
]
`;

exports[`table_voice_keys ビ.length 1 ~ 3 1`] = `"ヒ"`;

exports[`table_voice_keys ビ.length 1 ~ 3 2`] = `
Array [
  "ヒ",
  "ビ",
  "ピ",
]
`;

exports[`table_voice_keys ピ.length 1 ~ 3 1`] = `"ヒ"`;

exports[`table_voice_keys ピ.length 1 ~ 3 2`] = `
Array [
  "ヒ",
  "ビ",
  "ピ",
]
`;

exports[`table_voice_keys フ.length 1 ~ 3 1`] = `"フ"`;

exports[`table_voice_keys フ.length 1 ~ 3 2`] = `
Array [
  "フ",
  "ブ",
  "プ",
]
`;

exports[`table_voice_keys ブ.length 1 ~ 3 1`] = `"フ"`;

exports[`table_voice_keys ブ.length 1 ~ 3 2`] = `
Array [
  "フ",
  "ブ",
  "プ",
]
`;

exports[`table_voice_keys プ.length 1 ~ 3 1`] = `"フ"`;

exports[`table_voice_keys プ.length 1 ~ 3 2`] = `
Array [
  "フ",
  "ブ",
  "プ",
]
`;

exports[`table_voice_keys ヘ.length 1 ~ 3 1`] = `"ヘ"`;

exports[`table_voice_keys ヘ.length 1 ~ 3 2`] = `
Array [
  "ヘ",
  "ベ",
  "ペ",
]
`;

exports[`table_voice_keys ベ.length 1 ~ 3 1`] = `"ヘ"`;

exports[`table_voice_keys ベ.length 1 ~ 3 2`] = `
Array [
  "ヘ",
  "ベ",
  "ペ",
]
`;

exports[`table_voice_keys ペ.length 1 ~ 3 1`] = `"ヘ"`;

exports[`table_voice_keys ペ.length 1 ~ 3 2`] = `
Array [
  "ヘ",
  "ベ",
  "ペ",
]
`;

exports[`table_voice_keys ホ.length 1 ~ 3 1`] = `"ホ"`;

exports[`table_voice_keys ホ.length 1 ~ 3 2`] = `
Array [
  "ホ",
  "ボ",
  "ポ",
]
`;

exports[`table_voice_keys ボ.length 1 ~ 3 1`] = `"ホ"`;

exports[`table_voice_keys ボ.length 1 ~ 3 2`] = `
Array [
  "ホ",
  "ボ",
  "ポ",
]
`;

exports[`table_voice_keys ポ.length 1 ~ 3 1`] = `"ホ"`;

exports[`table_voice_keys ポ.length 1 ~ 3 2`] = `
Array [
  "ホ",
  "ボ",
  "ポ",
]
`;
